src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -179,7 +180,53 @@ export class SearchLogGroupWizard extends Wizard<SearchLogGroupWizardResponse> {
         })
 
         this.form.submenuResponse.bindPrompter(createRegionSubmenu)
-        this.form.filterPattern.bindPrompter(({ submenuResponse }) => createFilterpatternPrompter(submenuResponse!.data))
+        this.form.filterPattern.bindPrompter(({ submenuResponse }) =>
+            createFilterpatternPrompter(submenuResponse!.data)
+        )
         this.form.timeRange.bindPrompter(() => new TimeFilterSubmenu())
     }
 }
+
+export class JumpToStream implements vscode.DefinitionProvider {
+    private registry: LogStreamRegistry
+
+    public constructor(registry: LogStreamRegistry) {
+        this.registry = registry
+    }
+    async provideDefinition(
+        document: vscode.TextDocument,
+        position: vscode.Position,
+        token: vscode.CancellationToken
+    ): Promise<vscode.Definition | vscode.LocationLink[] | undefined> {
+        const activeUri = document.uri
+        const logGroupInfo = parseCloudWatchLogsUri(activeUri).logGroupInfo
+        const curLine = document.lineAt(position.line)
+        let streamUri: vscode.Uri
+        try {
+            const streamIDMap = this.registry.getStreamIdMap(activeUri)
+            const streamID = streamIDMap?.get(curLine.lineNumber)
+            if (!streamID) {
+                throw new Error(`cwl: No streamID found for stream with uri ${activeUri}`)
```

Comment:
Presumably this should only happen when looking at a single log stream, or if the streamIDMap hadn't been set before this function was called. Should I fail if this is a stream or if the streamIDMap is empty earlier rather than combining those two cases here?

Reply:
> Should I generally try to prioritize throwing very specific errors rather than trying to minimize the amount of code?

There's not really a good answer to this question as it is based off application requirements. Informative error messages is a very good thing for maintainability but it does increase development time. I wouldn't spend a bunch of time crafting error messages if you're not confident about what the future application requirements might be.

I think this specific assertion is good but I'm just pointing out that failing early tends to create code that is easier to reason about.

Reply:
That makes sense. Yeah I was trying to not have to have two different `if` statements to catch possible errors, but this meant that an error went unchecked for longer than it needed to. I adjusted it to catch the error earlier on.